### ISSUE TITLE
kata-runtime installation fails due to unsatisfiable vanilla-qemu dependency

### ISSUE THREAD

**Issue by sray** [kata-containers/documentation#685]
# Description of problem

Executing:
```bash
BRANCH=stable-1.11 bash -c "$(curl -fsSL https://raw.githubusercontent.com/kata-containers/tests/master/cmd/kata-manager/kata-manager.sh) install-packages"
```
on Debian 10 results in:
```bash
The following packages have unmet dependencies:
 kata-runtime : Depends: qemu-vanilla (= 4.1.1+git.99c5874a9b-5) but 4.1.1+git.99c5874a9b-2 is to be installed
```

# Expected result

qemu-vanilla=4.1.1+git.99c5874a9b-2 should be sufficient - the commit hash is the same, so the package should have the exact same content. (Wondering, if this caused by a typo in the dependencies somewhere?)

# Actual result

see Description of problem

**Comment by sray** [comment]
Just found that issue in the packaging repo: https://github.com/kata-containers/packaging/issues/1051
Is this related?

**Comment by devimc** [comment]
I think so, since I don't see Debian 10 here https://build.opensuse.org/package/show/home:katacontainers:releases:x86_64:master/qemu-vanilla, I guess it's time to check if QEMU 5 works on Debian 10

**Comment by sray** [comment]
@devimc there is Debian 10 build inside http://download.opensuse.org/repositories/home:/katacontainers:/releases:/x86_64:/stable-1.11/ and this references qemu-vanilla in the kata-runtime package, but the difference between the unmet dependency and the available dependencies is only the -5 / -2:
```bash
4.1.1+git.99c5874a9b-5 #kata runtime depends on -5
4.1.1+git.99c5874a9b-2 #debian 10 provides -2

So, the -5/-2 indicates the QEMU version?
And can I install kata containers in Debian 10 with another runtime instead of vanilla-qemu?

**Comment by devimc** [comment]
> So, the -5/-2 indicates the QEMU version?

good catch, I have no idea why the -2 in qemu-vanilla
cc @rhafer @jodh-intel 

> And can I install kata containers in Debian 10 with another runtime instead of vanilla-qemu?

vanilla-qemu is not a runtime, it's the hypervisor, kata support other hypervisors like , [cloud-hypervisor](https://github.com/cloud-hypervisor/cloud-hypervisor), [firecracker](https://github.com/firecracker-microvm/firecracker) and [ACRN](https://github.com/projectacrn/acrn-hypervisor), but if you still want to use QEMU, install it form the debian repos (`apt-get install qemu-system-${arch}`) and change the path in the configuration file `/usr/share/default/kata-containers/configuration.toml`

**Comment by sray** [comment]
@devimc thanks for clarifying! I was thinking of "vm runtime" ;-) hypervisor is the better term ... way too many things are called runtime these days and I have been reading way too much documentation the last couple of days :-D

Regarding firecracker, I would go straight for this, but the dependency problem with qemu-vanilla when installing kata-runtime persists, unless I compile kata-runtime myself from source? (which in turn excludes the installation from simple package manager updates - is there a guide on how to manually update a from-source-installation somewhere?)

**Comment by devimc** [comment]
> Regarding firecracker, I would go straight for this, but the dependency problem with qemu-vanilla when installing kata-runtime persists, unless I compile kata-runtime myself from source? (which in turn excludes the installation from simple package manager updates - is there a guide on how to manually update a from-source-installation somewhere?)

try adding ` --ignore-missing` to the `apt-get` command

> go get -d -u github.com/kata-containers/runtime
package github.com/kata-containers/runtime: no Go files in /root/go/src/github.com/kata-containers/runtime

this is a warning, not an error


maybe you want to take a look to the kata containers snap https://snapcraft.io/kata-containers

**Comment by sray** [comment]
@devimc --ignore-missing seems to have no effect, I still can't install kata-runtime using: `apt-get --ignore-missing kata-runtime` results in same depdency error as before.

Continuing the build from source with make despite the warning works. If there is a migration guide to switch to the package manager version of kata-runtime later, once it works again, I would consider using it in production already. If not, it's probably better to stay on debian 9 and wait for the issue to be resolved in the package dependencies before doing a dist-upgrade + a kata-runtime update.

Regarding snaps, there is quite some critism in the different linux communities, so I am hesistant to use it - and https://forum.snapcraft.io/t/disabling-automatic-refresh-for-snap-from-store/707/282 is still an ongoing discussion (which would make me not use it in production environments yet). Besides, installing your stack at least once using the existing documentation teaches quite a bit about dependencies and helps in tracking down problems later on. From the snap website it is also hard to see, what's inside the snap (e.g. is firecracker also in there? If not, how is it possible to use different hypervisors using the snap?).

**Comment by devimc** [comment]
@sray thanks for the link, yeah maybe you should wait a little bit more before using Debian 10 + Kata in production, maybe @amshinde or @chavafg can help us with the packaging issue

**Comment by rhafer** [comment]
> good catch, I have no idea why the -2 in qemu-vanilla
> cc @rhafer @jodh-intel

I am not entirely sure what that is as well. It seems to be a the Debian dsc equivalent for the `Release` Tag in RPM spec files. Though I am not sure why the deb package are including that in their dependencies. The specfiles don't. See:
https://github.com/kata-containers/packaging/blob/master/obs-packaging/runtime/kata-runtime.dsc-template#L22
vs:
https://github.com/kata-containers/packaging/blob/master/obs-packaging/runtime/kata-runtime.spec-template#L31

Also note that the latest build of qemu-vanilla for debian 10 failed: https://build.opensuse.org/package/live_build_log/home:katacontainers:releases:x86_64:stable-1.11/qemu-vanilla/Debian_10/x86_64


That one would have gotten the `-5` in its version string. At least according to the `dsc` file (https://build.opensuse.org/package/view_file/home:katacontainers:releases:x86_64:stable-1.11/qemu-vanilla/qemu-vanilla.dsc?expand=1)

**Comment by devimc** [comment]
thanks @rhafer , I think you're right it's using qemu `-2` because qemu `-5` failed, I hope this issue gets fixed in kata 1.12 since it uses QEMU 5

**Comment by ariel-adam** [comment]
The issue has been fixed as far as we know.
